pynecone/components/component.py

```diff
@@ -158,7 +158,7 @@ def _create_event_chain(
             ValueError: If the value is not a valid event chain.
         """
         # If it's already an event chain, return it.
-        if isinstance(value, Var):
+        if isinstance(value, EventChain):
```

Comment:
I think here the code is correct but the comment is incorrect.

Can we revert the code, but fix the comment to say:
```python
# If it's a var, return it.
```

Reply:
Maybe in addition, we need the `EventChain` check? But we definitely need this current one. What code did you run that raised the `ValueError`?

Reply:
EventChain check is for CustomComponent. 

I run the same code from #310.
```python
class State(pc.State):
    some_var: bool

def index():
    return pc.box(pc.icon(tag="AddIcon", on_click=State.some_var))
```

some_var is passed as a BaseVar, it is a child class of Var, so it is caught at first if statement.

Reply:
Can you explain more about why we definitely need the current one when we create EventChain? BaseVar shouldn't be returned in my opinion.

Reply:
Ahh I see - you're right this is causing the other bug it looks like. We need this line for the [CustomComponent](https://github.com/pynecone-io/pynecone/blob/a4ac2979c815d417e84699c2a764dbefa873776e/pynecone/components/component.py#L415) feature that I'm working on (it's not released yet, but it will allow you to create [memoized React components ](https://beta.reactjs.org/reference/react/memo)for better performance).

But I think in adding that line it caused this other bug. Let me think a bit more on how to resolve this.

Reply:
Maybe we can add a line like
```python
if isinstance(self, CustomComponent):
    if isinstance(value, Var):
        return value
```
Then it should fix this bug, but still work for the new feature.

Reply:
Ok. Should I change the code as you suggest? or just close this pr and wait for your new feature?

Reply:
I just did a quick fix as you suggest.